react/features/e2ee/middleware.js

```diff
@@ -35,6 +46,69 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
         dispatch(unregisterSound(E2EE_ON_SOUND_ID));
         break;
 
+    case PARTICIPANT_UPDATED:
+    case PARTICIPANT_JOINED: {
```

Comment:
If we move the middleware implementation directly into the reducer we won't need to dispatch new actions. Generally this would be optimisation in the cases where we need to dispatch something because no additional middlewares and subscriber will called. Also all the mapStateToProps will be called once.

Maybe we should move it to the reducer directly, WDYT?

Reply:
Isn't the reducer to change only local state, we need to state of other features(participants) and that's why this is in the middleware.

Reply:
Yes! I got confused :( Ignore the comment.